crates/bevy_reflect/src/map.rs

```diff
@@ -455,6 +455,26 @@ impl<'a> Iterator for MapIter<'a> {
     }
 }
 
+impl FromIterator<(Box<dyn Reflect>, Box<dyn Reflect>)> for DynamicMap {
+    fn from_iter<I: IntoIterator<Item = (Box<dyn Reflect>, Box<dyn Reflect>)>>(items: I) -> Self {
+        let mut dynamic_map = Self::default();
+        for (key, value) in items.into_iter() {
+            dynamic_map.insert_boxed(key, value);
+        }
+        dynamic_map
+    }
+}
+
+impl<K: Reflect, V: Reflect> FromIterator<(K, V)> for DynamicMap {
+    fn from_iter<I: IntoIterator<Item = (K, V)>>(items: I) -> Self {
+        let mut map = Self::default();
```

Comment:
Nit: this is called `map` here and `dynamic_map` above but they are both `DynamicMap`s